### PR TITLE
docs: copy state to props is a bad practice

### PR DESCRIPTION
**Pull request by GuichiZhao**
The original code copy the props the state which is not necessary at all, it will only confuse the reader!
Also, it considered being a bad practice: https://overreacted.io/writing-resilient-components/

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/form/demo/customized-form-controls.md](https://github.com/GuichiZhao/ant-design/blob/patch-1/components/form/demo/customized-form-controls.md)

### CODE REVIEW

**Comment by GuichiZhao** [comment]
不是很会用你们的pull request，不过有没有人能看一下啊，把那么低级的编程错误堂而皇之放在官网真的不怕丢人的吗。。。。。

**Comment by afc163** [comment]
Would you mind check the broken ci above and fix it?

**Comment by afc163** [comment]
You can try `npm run lint:demo` to check lint.

**Comment by GuichiZhao** [comment]
@afc163 all the check passed

File: components/form/demo/customized-form-controls.md [hunk 1]
```diff
@@ -31,117 +31,89 @@ import { Form, Input, Select, Button } from 'antd';
...
-    }
-    return null;
-  }
-
-  constructor(props) {
-    super(props);
+    handleNumberChange = e => {
+        const number = parseInt(e.target.value || 0, 10);
+        if (isNaN(number)) {
+            return;
```
Comment (afc163) [inline components/form/demo/customized-form-controls.md]: Could you modify indent to 2 spaces?

File: components/form/demo/customized-form-controls.md [hunk 2]
```diff
@@ -27,73 +27,44 @@ Customized or third-party form controls can be used in Form, too. Controls must
...
       onChange({
-        ...this.state,
+        ...this.props.value,
         ...changedValue,
       });
     }
   };
 
   render() {
     const { size } = this.props;
```
Comment (afc163) [inline components/form/demo/customized-form-controls.md]:
```suggestion
    const { size, value } = this.props;
```

**Pull request merged**